Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -39,9 +40,11 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
 
     private func commonInit() {
         super.delegate = self
+        super.addTarget(self, action: "SELtextDidChange:", forControlEvents: UIControlEvents.EditingChanged)
         notifyTextChanged = debounce(0.1, {
             if self.editing {
-                self.autocompleteDelegate?.autocompleteTextField(self, didEnterText: self.text)
+                let enteredText = self.text.substringToIndex(advance(self.text.startIndex, self.enteredTextLength, self.text.endIndex))
+                self.autocompleteDelegate?.autocompleteTextField(self, didEnterText: enteredText)
```

Comment:
@thebnich  You always know what is best :).
 Changed `enteredTextLength ` to `entredText` working good.
As we are not calculating `enteredTextLength` when in completion mode and if user enters some text in between than it will be calculated in `SELtextDidChange `. Thanks for your suggestion.